src/components/tx/SignOrExecuteForm/SignOrExecuteForm.test.tsx

```diff
@@ -0,0 +1,209 @@
+import { act, fireEvent, render } from '@/tests/test-utils'
+import SignOrExecuteForm from '@/components/tx/SignOrExecuteForm/index'
+import { SafeSignature, SafeTransaction } from '@gnosis.pm/safe-core-sdk-types'
+import * as useSafeInfoHook from '@/hooks/useSafeInfo'
+import * as useGasLimitHook from '@/hooks/useGasLimit'
+import * as txSender from '@/services/tx/txSender'
+import * as wallet from '@/hooks/wallets/useWallet'
+import { SafeInfo, TransactionDetails } from '@gnosis.pm/safe-react-gateway-sdk'
+import { waitFor } from '@testing-library/react'
+import { ConnectedWallet } from '@/services/onboard'
+import * as safeCoreSDK from '@/hooks/coreSDK/safeCoreSDK'
+import Safe from '@gnosis.pm/safe-core-sdk'
+
+const createSafeTx = (): SafeTransaction => {
+  return {
+    data: {
+      to: '0x0000000000000000000000000000000000000000',
+      value: '0x0',
+      data: '0x',
+      operation: 0,
+      nonce: 100,
+    },
+    signatures: new Map([]),
+    addSignature: function (sig: SafeSignature): void {
+      this.signatures.set(sig.signer, sig)
+    },
+    encodedSignatures: function (): string {
+      return Array.from(this.signatures)
+        .map(([, sig]) => {
+          return [sig.signer, sig.data].join(' = ')
+        })
+        .join('; ')
+    },
+  } as SafeTransaction
+}
+
+describe('SignOrExecuteForm', () => {
+  let mockSDK
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+
+    mockSDK = {
+      isModuleEnabled: jest.fn(() => false),
+      createTransaction: jest.fn(() => 'asd'),
+      getTransactionHash: jest.fn(() => '0x10'),
+    } as unknown as Safe
+
+    jest.spyOn(safeCoreSDK, 'getSafeSDK').mockReturnValue(mockSDK)
+    jest.spyOn(useSafeInfoHook, 'default').mockImplementation(() => ({
+      safe: {
+        nonce: 100,
+      } as SafeInfo,
+      safeAddress: '0x123',
+      safeError: undefined,
+      safeLoading: false,
+      safeLoaded: true,
+    }))
+    jest.spyOn(useGasLimitHook, 'default').mockReturnValue({
+      gasLimit: undefined,
+      gasLimitError: undefined,
+      gasLimitLoading: false,
+    })
+    jest.spyOn(wallet, 'default').mockReturnValue({} as ConnectedWallet)
+    jest
+      .spyOn(txSender, 'dispatchTxProposal')
+      .mockImplementation(jest.fn(() => Promise.resolve({ txId: '0x12' } as TransactionDetails)))
+
+    jest.spyOn(txSender, 'dispatchTxExecution').mockImplementation(jest.fn())
+  })
+
+  it('displays decoded data if there is a tx', () => {
+    const mockTx = createSafeTx()
+    const result = render(<SignOrExecuteForm isExecutable={true} onSubmit={jest.fn} safeTx={mockTx} txId="mockTxId" />)
+
+    expect(result.getByText('Transaction details')).toBeInTheDocument()
+  })
+
+  it('doesnt display decoded data if there is no tx', () => {
+    const result = render(<SignOrExecuteForm isExecutable={true} onSubmit={jest.fn} safeTx={undefined} />)
+
+    expect(result.queryByText('Transaction details')).not.toBeInTheDocument()
```

Comment:
Should we also test for contents of the details? To make sure not just the label is there?

Reply:
Good idea! I suggest we add those tests for the `<DecodedTx>` component in a separate PR as its a bit cleaner to test it in isolation without the `SignOrExecuteForm`.